src/Cardano/Prelude/GHC/Heap/NormalForm/Classy.hs

```diff
@@ -182,6 +184,36 @@ noUnexpectedThunksInKeysAndValues ctxt =
                             , noUnexpectedThunks ctxt v
                             ])
 
+unsafeNoUnexpectedThunks :: NoUnexpectedThunks a => a -> Maybe String
```

Comment:
You almost certainly want an `{-# NOINLINE unsafeNoUnexpectedThunks #-}` on that, because if this gets inlined you will get weird behaviour.

Reply:
Thanks, I hadn't known this. For my own future reference (and anyone reading) this is documented [here](https://hackage.haskell.org/package/base-4.12.0.0/docs/System-IO-Unsafe.html#v:unsafePerformIO).